libsolidity/interface/FileReader.cpp

```diff
@@ -81,7 +80,7 @@ ReadCallback::Result FileReader::readFile(string const& _kind, string const& _pa
 
 		// NOTE: we ignore the FileNotFound exception as we manually check above
 		auto contents = readFileAsString(canonicalPath.string());
-		m_sourceCodes[path.generic_string()] = contents;
+		m_sourceCodes[_sourceUnitName] = contents;
```

Comment:
This is the part that fixes the bug. The rest of the PR is just fluff like tests, docs, refactors, etc.

Reply:
if you pass the `_sourceUnitName` as value parameter you then could move that in here. Would this make sense?

Reply:
Yeah, but this is a callback so I think it has to follow the original signature exactly. I.e. I can change argument name but not necessarily the type.